### ISSUE TITLE
Color alphabet for lane connector

### ISSUE THREAD

**Issue by aubergine10** [krzychu124/Cities-Skylines-Traffic-Manager-President-Edition#287]
Jotting down notes on research for new lane connector color alphabet.

The existing color alphabet consists of 48 colours as shown below:

![colors](https://user-images.githubusercontent.com/1386719/56445290-74a4a400-62f4-11e9-9617-2f101d040ef8.jpg)

[Research suggests](https://aic-color.org/resources/Documents/jaic_v5_06.pdf) a color alphabet should be 26 colors or less if there are no limits on which colors can be used.

As we are working with roads which are generally grey to black in color (and sometimes red or green for bus and bike lanes), we should avoid dark/deep colors, and also shades of grey.

![lightness](https://user-images.githubusercontent.com/1386719/56445571-cbf74400-62f5-11e9-907f-8a47d3ef127c.jpg)

Any reds and greens (and potentially blues & yellows which might become prevalent due to Network Skins 2) should be distinct from common road surface colors used in the game. As we have plans to apply special meaning to white in later updates, that color must be avoided.

Some standard color alphabets are shown below (note that several of the colors in each are not suitable for lane connectors):

![color alphabet](https://user-images.githubusercontent.com/1386719/56445661-4cb64000-62f6-11e9-880d-a00bb451afc3.jpg)

I think we could potentially use A-P on the following alphabet:

![alphabet](https://user-images.githubusercontent.com/1386719/56445856-483e5700-62f7-11e9-8be9-b4fadbddd226.jpg)

And some samples of colors used on transport maps around the world (note the lack of contrast between orange and yellow):

![transit-line-colors-of-US-and-Canada](https://user-images.githubusercontent.com/1386719/56445685-6a83a500-62f6-11e9-9bad-3bca84e26f3d.jpg)

Sample of vivid colors that would work well on both light and dark road surfaces:

![BoroughHall](https://user-images.githubusercontent.com/1386719/56445731-a74f9c00-62f6-11e9-8f83-630fa66194ea.jpg)

We should also consider [color blindness](https://www.color-blindness.com/coblis-color-blindness-simulator/), and [prevalence](https://www.colour-blindness.com/general/prevalence/) of each type.

Ordering the colors based on color blindness prevalence would potentially allow us to offer accessibility features for color blind users:

* For 100% coverage we'd be limited to yellow and blue
* For 99.99% coverage we could add maroon, lavender, orange and pink
* For 99% coverage we could add brown, "red" (more of a vibrant pink), beige, mint, green, cyan, teal, and magenta
* For 95% coverage we could add olive, lime, apricot and purple

That would be a total of 18 colors, with options to reduce palette to subsets of those colors for accessibility purposes.

I still have some stuff to check, but should have an initial color alphabet ready within the next day or two.

**Comment by aubergine10** [comment]
This is derived from an alphabet by Sasha Trubetskoy:

![palette](https://user-images.githubusercontent.com/1386719/56447300-52fcea00-62ff-11e9-9c99-368446f8470b.jpg)

Note that whatever palette we decide on will need testing with and without daylight classic mod. I suspect yellow/orange colors will suffer with vanilla sun tint.

**Comment by aubergine10** [comment]
Quick progress update - taking much longer than expected.

I'm focusing on the central band of the image above, finding nicer colors to use, and then deriving a second shade of each of those colors whilst simultaneously testing for 8 types of color blindness.

I'm trying to avoid having 9 color palettes, although part of my brain is wondering if it would be better to do the opposite.

**Comment by aubergine10** [comment]
I'm resorting to using HTML to test colors in order to get faster turnaround on iterations....

Does anyone have vanilla pavement and asphalt textures (I want to set html background image to those for testing colors on top of them).

**Comment by aubergine10** [comment]
slowly getting there...

![swatches](https://user-images.githubusercontent.com/1386719/57233695-0d3a6400-7017-11e9-8417-dfa5834bcc25.jpg)

**Comment by aubergine10** [comment]
Can now apply color blindness effects directly to the palette in browser:

![cb](https://user-images.githubusercontent.com/1386719/57282552-33a6e080-70a5-11e9-94fe-1e31130e4352.jpg)

One important note from earlier information is that we will have to ensure bold colors to accommodate users with Blue Cone Monochromacy (Acrhomatomaly) which causes colors to be faded / less saturated.

![toggle](https://user-images.githubusercontent.com/1386719/57283007-1c1c2780-70a6-11e9-97de-5df376e16617.jpg)

I'm now working on tooling to allow colors to be toggled per type of color blindness (see above) and then it should just be a case of iterating the swatches to output C code for color palettes. Oh, and actually deciding on final colors which is probably going to take another day or so, but will endeavour to get it done today.

**Comment by aubergine10** [comment]
Another note: Assuming we are using unlit shader, in-game color distortions (sun tint, LUT, day/night, etc) wont affect the color palette :)
